actpool/actqueue.go

```diff
@@ -103,17 +104,24 @@ func NewActQueue(ap *actPool, address string, ops ...ActQueueOption) ActQueue {
 	return aq
 }
 
-// Overlap returns whether the current queue contains the given nonce
+// Overlap returns whether the current queue contains the given act
 func (q *actQueue) Overlaps(act action.SealedEnvelope) bool {
-	_, exist := q.items[act.Nonce()]
-	return exist
+	if actInPool, exist := q.items[act.Nonce()]; exist && hex.EncodeToString(actInPool.Signature()) == hex.EncodeToString(act.Signature()) {
```

Comment:
how does this work?

Reply:
Same acts whose signatures and gas prices are same will be rejected in https://github.com/Liuhaai/iotex-core/blob/issue%232801/actpool/actpool.go#L374. The act with higher gas price will be put in  https://github.com/Liuhaai/iotex-core/blob/issue%232801/actpool/actqueue.go#L118.

Reply:
also had the same question, why need to check signature is equal now?

in line 123 below, tx with higher gas will replace the existing one in the queue, so for each `nonce`, there is ONLY 1 tx in the queue, right? so no need to check signature is equal?

Reply:
> also had the same question, why need to check signature is equal now?
> 
> in line 123 below, tx with higher gas will replace the existing one in the queue, so for each `nonce`, there is ONLY 1 tx in the queue, right? so no need to check signature is equal?

Sig checking is bad idea, It will be removed